compiler/il/AliasSetInterface.hpp

```diff
@@ -284,57 +278,93 @@ class TR_SymAliasSetInterface : public TR_AliasSetInterface<TR_SymAliasSetInterf
     (symRef, isDirectCall, includeGCSafePoint) {}
 
    TR_UseDefAliasSetInterface(bool shares_symbol,
                               TR::SymbolReference *symRef,
                               bool isDirectCall = false,
                               bool includeGCSafePoint = false) :
-  TR_SymAliasSetInterface<useDefAliasSet>
+  TR_AliasSetInterface<useDefAliasSet>
     (shares_symbol, symRef, isDirectCall, includeGCSafePoint) {}
 };
 
-struct TR_UseOnlyAliasSetInterface: public TR_SymAliasSetInterface<UseOnlyAliasSet> {
+struct TR_UseOnlyAliasSetInterface: public TR_AliasSetInterface<UseOnlyAliasSet> {
   TR_UseOnlyAliasSetInterface(TR::SymbolReference *symRef,
                               bool isDirectCall = false,
                               bool includeGCSafePoint = false) :
-  TR_SymAliasSetInterface<UseOnlyAliasSet>
+  TR_AliasSetInterface<UseOnlyAliasSet>
     (symRef, isDirectCall, includeGCSafePoint) {}
 };
 
-template <uint32_t _aliasSetType> inline
-void TR_SymAliasSetInterface<_aliasSetType>::setAlias_impl(TR::SymbolReference *symRef2, bool value, bool isDirectCall, bool includeGCSafePoint)
+/*
+ * Member function definitions of class TR_AliasSetInterface
+ */
+template <> inline
+TR_BitVector *TR_AliasSetInterface<useDefAliasSet>::getTRAliases_impl(bool isDirectCall, bool includeGCSafePoint)
+   {
+   if(_symbolReference)
+      {
+      if(_shares_symbol)
+         return _symbolReference->getUseDefAliasesBV(isDirectCall, includeGCSafePoint);
+
+      else
+         {
+         //if symbol refereance cannot have alias 
+         TR::Compilation *comp = TR::comp();
+         TR_BitVector *bv = NULL;
+
+         bv = new (comp->aliasRegion()) TR_BitVector(comp->getSymRefCount(), comp->aliasRegion(), growable);
+         bv->set(_symbolReference->getReferenceNumber());
+
+         return bv;
+         }
+      }
+   else
+      //if there is no symbol reference associated with the node, we return an empty bit container
+      return NULL;
+   }
+
+template <> inline
+TR_BitVector *TR_AliasSetInterface<UseOnlyAliasSet>::getTRAliases_impl(bool isDirectCall, bool includeGCSafePoint)
+   {      
+   if(!_symbolReference)
+      //if there is no symbol reference, then we return an empty bit container
+      return NULL;
+      
+   return _symbolReference->getUseonlyAliasesBV(TR::comp()->getSymRefTab());
+   }
+template <uint32_t _AliasSetInterface> inline
+void TR_AliasSetInterface<_AliasSetInterface>::setAlias_impl(TR::SymbolReference *symRef2, bool value, bool isDirectCall, bool includeGCSafePoint)
    {
    TR::Compilation *comp = TR::comp();
    if (symRef2 == NULL)
       return;
 
-   TR_ASSERT(_aliasSetType == useDefAliasSet || _aliasSetType ==  UseOnlyAliasSet, "failed: can only call setAlias_impl for UseOnly or useDef presently");
+   TR_ASSERT(_AliasSetInterface == useDefAliasSet || _AliasSetInterface ==  UseOnlyAliasSet, "failed: can only call setAlias_impl for UseOnly or useDef presently");
```

Comment:
should be TR_ASSERT_FATAL IMO to make sure this is checked...

Reply:
Have changed to `TR_ASSERT_FATAL` in 091c9f0